### PR TITLE
48634_course_assign_to_supervisor_&_trainee(both)

### PR DESCRIPTION
**Pull request by noor-framgia**

### CODE REVIEW

File: app/models/course.rb [hunk 1]
```diff
@@ -4,6 +4,8 @@ class Course < ActiveRecord::Base
 
   has_many :user_courses, dependent: :destroy
   has_many :users, through: :user_courses
+  has_many :trainees, -> {trainees}, class_name: "User", through: :user_courses
+  has_many :trainees, -> {supervisors}, class_name: "User", through: :user_courses
```
Comment (manazirahsan) [inline app/models/course.rb]: One should be :supervisor.

Comment (dieunb) [inline app/models/course.rb]: duplicated?

File: app/views/supervisor/users_courses/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,58 @@
+<% provide :title, t("add_edit_enrollment") %>
+<h1><%= t(:add_edit_enrollment) %></h1>
+<h3><%= t(:course) %>: <%= @course.name %></h3> <br/>
+
+<%= form_for @course, url: supervisor_users_course_path(@course) do |f| %>
+  <%= render 'shared/error_messages', object: f.object %>
```
Comment (dieunb) [inline app/views/supervisor/users_courses/show.html.erb]: double quote

File: app/models/user.rb [hunk 3]
```diff
@@ -29,6 +28,12 @@ class User < ActiveRecord::Base
   has_many :user_subjects
   has_many :course_subjects, through: :user_subjects
 
+  scope :supervisors, -> {supervisor}
+  scope :trainees, -> {trainee}
+
+  enum role: [:supervisor, :trainee]
+  ROLE = { supervisor: 'supervisor', trainee: 'trainee' }
```
Comment (dieunb) [inline app/models/user.rb]: convention + double quote pls

File: app/views/supervisor/users_courses/show.html.erb [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+      <table class="table">
+        <thead>
+        <tr>
+          <th><%= t(:select) %></th>
+          <th><%= t(:name) %></th>
+          <th><%= t(:email) %></th>
+        </tr>
+        </thead>
+        <tbody>
+        <%= collection_check_boxes :course, :user_ids, @supervisors, :id, :name do |add_supervisor| %>
```
Comment (dieunb) [inline app/views/supervisor/users_courses/show.html.erb]: indent

File: app/views/supervisor/users_courses/show.html.erb [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+      <table class="table">
+        <thead>
+        <tr>
+          <th><%= t(:select) %></th>
+          <th><%= t(:name) %></th>
+          <th><%= t(:email) %></th>
+        </tr>
+        </thead>
+        <tbody>
+        <%= collection_check_boxes :course, :user_ids, @trainees, :id, :name do |add_trainee| %>
```
Comment (dieunb) [inline app/views/supervisor/users_courses/show.html.erb]: indent

**Comment by manazirahsan** [comment]
okey

File: app/controllers/supervisor/users_courses_controller.rb [hunk 6]
```diff
@@ -0,0 +1,27 @@
+class Supervisor::UsersCoursesController < ApplicationController
```
Comment (dieunb) [inline app/controllers/supervisor/users_courses_controller.rb]: we don't have users_courses_controller in supervisor.
To assign trainee to course, should create `assign_trainees_controller` and use nested resourse as below:
```
resourses :courses do 
  resourse :assign_trainee # without "s"
end
```

The same with assign supervisor

Comment (noor-framgia) [inline app/controllers/supervisor/users_courses_controller.rb]: we have that controller, do you want that I change the controller to assign_trainee and assign_supervisor controller under courses?

Comment (dieunb) [inline app/controllers/supervisor/users_courses_controller.rb]: ok, do it as you think!

File: app/views/supervisor/users_courses/show.html.erb [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+  </ul>
+
+  <div class="tab-content">
+    <div id="home" class="tab-pane fade in active">
+      <table class="table">
+        <tbody>
+          <%= collection_check_boxes :course, :user_ids, User.supervisors, :id, :name do |added_supervisor| %>
+            <%= added_supervisor.label class:"label-checkbox" do%>
+             <%= added_supervisor.check_box + added_supervisor.text %>
+            <%end%><br>
```
Comment (MuhammadTamzid) [inline app/views/supervisor/users_courses/show.html.erb]: indent
<% end %>

File: app/views/supervisor/users_courses/show.html.erb [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+        <tbody>
+      </table>
+    </div>
+    <div id="menu1" class="tab-pane fade">
+      <table class="table">
+        <tbody>
+          <%= collection_check_boxes :course, :user_ids, User.trainees, :id, :name do |added_trainee| %>
+            <%= added_trainee.label class:"label-checkbox" do %>
+             <%= added_trainee.check_box + added_trainee.text %>
+            <%end%><br>
```
Comment (MuhammadTamzid) [inline app/views/supervisor/users_courses/show.html.erb]: indent
<% end %>

File: app/views/supervisor/users_courses/show.html.erb [hunk 9]
```diff
@@ -0,0 +1,42 @@
+<% provide :title, t(:enrollment) %>
+<h1><%= t(:enrollment) %></h1>
+<h3><%= t(:course) %>: <%= @course.name %></h3> <br/>
+
+<%= form_for @course, url: supervisor_users_course_path(@course) do |f| %>
+  <%= render "shared/error_messages", object: f.object %>
+
+  <% hidden_field_tag :id, @course.id %>
```
Comment (MuhammadTamzid) [inline app/views/supervisor/users_courses/show.html.erb]: why? 
is it necessary?

**Pull request merged**